grammars/javascript.cson

```diff
@@ -1806,72 +1806,87 @@
           (?:(?<=@param)|(?<=@arg)|(?<=@argument)|(?<=@type))
           \\s+
           ({(?:
-            \\* |                                        # {*} any type
-            \\? |                                        # {?} unknown type
-
-            (?:                                          # Check for a prefix
-              \\? |                                      # {?string} nullable type
-              !   |                                      # {!string} non-nullable type
-              \\.{3}                                     # {...string} variable number of parameters
-            )?
+            \\* |                                       # {*} any type
+            \\? |                                       # {?} unknown type
 
             (?:
-              \\(                                        # Opening bracket of multiple types with parenthesis {(string|number)}
-                [a-zA-Z_$]+
+              (?:                                       # Check for a prefix
+                \\? |                                   # {?string} nullable type
+                !   |                                   # {!string} non-nullable type
+                \\.{3}                                  # {...string} variable number of parameters
+              )?
+
+              (?:
                 (?:
-                  (?:
-                    [\\w$]*
-                    (?:\\[\\])?                          # {(string[]|number)} type application, an array of strings or a number
-                  ) |
-                  \\.?<[\\w$]+(?:,\\s+[\\w$]+)*>         # {Array<string>} or {Object<string, number>} type application (optional .)
-                )
+                  function\\(                           # {function(string, number)} function type
+                    (?:
+                      \\s*
+                      [a-zA-Z_$]
+                      (?:\\s*,\\s*[\\w$])*
+                      \\s*
+                    )*
+                  \\)
+                )? |
```

Comment:
This `|` is in a bit of an awkward place - maybe move it to the empty line below or just remove that empty line.